kpet/data.py

```diff
@@ -327,6 +376,64 @@ def is_dir_valid(dir_path):
         """
         return os.path.isfile(dir_path + "/index.yaml")
 
+    def validate(self):
+        """
+        Check if any of the regexes are invalid.
+
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+        host_types = list()
+        trees = list()
+        arches = list()
+
+        for host_name in self.host_types or []:
+            host_types.append(host_name)
+        trees = list(self.trees.keys())
+        arches = self.arches
+
+        error = ("Configuration error: One (or more) of {match} - {what} " +
+                 "regexes\ndoesn't match any of the available {what}\n" +
+                 "In suite: {suite.description}\n" +
+                 "In case: {case.name}\n" +
+                 "The regex: {regex.pattern}\n" +
+                 "The avaliable {what}: {dataset}")
+
+        for suite in self.suites or []:
+            for case in suite.cases or []:
+                host_type_regex = case.host_type_regex or \
+                        suite.host_type_regex or \
+                        self.host_type_regex
+
+                match, values = case.dont_match.validate(Target(set(trees),
```

Comment:
Could you call the `values` variable something more meaningful? Like `mismatch_details`, or `mismatch_desc`?